spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/properties/bind/PropertySourcesPlaceholdersResolver.java

```diff
@@ -63,18 +64,22 @@ public Object resolvePlaceholders(Object value) {
 		return value;
 	}
 
-	protected String resolvePlaceholder(String placeholder) {
+	private String resolvePlaceholder(String placeholder) {
 		if (this.sources != null) {
 			for (PropertySource<?> source : this.sources) {
 				Object value = source.getProperty(placeholder);
 				if (value != null) {
-					return String.valueOf(value);
+					return convertResolvedValueToString(placeholder, value, source);
 				}
 			}
 		}
 		return null;
 	}
 
+	protected String convertResolvedValueToString(String placeholder, Object value, PropertySource<?> source) {
```

Comment:
I changed the method that we allow to override, making the previous one private. I thought it's not a problem since `resolvePlaceholder` was made protected only recently (commit eb045f1da13083cf2cd6f4d45184c252969be76c).